eml_parser/decode.py

```diff
@@ -118,8 +118,13 @@ def decode_string(string: bytes, encoding: typing.Optional[str]) -> str:
 
     if chardet:
         enc = chardet.detect(string)
-        if not (enc['confidence'] is None or enc['encoding'] is None) and not (enc['confidence'] == 1 and enc['encoding'] == 'ascii'):
-            value = string.decode(enc['encoding'], 'replace')
+        if not(enc['confidence'] is None or enc['encoding'] is None) and not(enc['confidence'] == 1 and enc['encoding'] == 'ascii'):
+            try:
+                value = string.decode(enc['encoding'], 'replace')
+            except LookupError:
+                if enc['encoding'] == 'VISCII':
```

Comment:
If I understand your change correctly, VISCII is similar to ASCII (wikipedia) but VISCII is not supported by python (checked) but chardet may detect text as VISCII.

Please change your condition to (unless we can be sure that it is always all-caps ?) :
```
if enc['encoding'].lower() == 'viscii':
```

and put it in front of the big **if** like so:

```
if enc['encoding'] is not None and enc['encoding'].lower() == 'viscii':
    value = string.decode('ascii', 'replace')
elif not (enc['confidence'] is None or e....
```

Also make sure to add a comment why you are testing for VISCII specifically.

I think it is a good idea to keep the try catch though and force decoding as ascii (just without the conditional).

Reply:
Changes have been applied as suggested 😀